plugins/module_utils/job.py

```diff
@@ -232,13 +240,24 @@ def _get_job_status(job_id="*", owner="*", job_name="*", dd_name=None, duration=
             job["ret_code"] = {}
             job["ret_code"]["msg"] = entry.status + " " + entry.rc
             job["ret_code"]["msg_code"] = entry.rc
-            # Why was this set to an empty string?
+
             job["ret_code"]["code"] = None
             if len(entry.rc) > 0:
                 if entry.rc.isdigit():
                     job["ret_code"]["code"] = int(entry.rc)
             job["ret_code"]["msg_text"] = entry.status
 
+            # this section only works on zoau 1.2.3 vvv
```

Comment:
So I went on a little rabbit hole for this, I just learnt [how Python compares strings when it's not just `=`](https://docs.python.org/3/tutorial/datastructures.html#comparing-sequences-and-other-types).

It turns out that if ZOAU decided to release a version 1.2.10, Python would evaluate `"1.2.10" > "1.2.3"` as `False` because it compares character by character. Testing a bit on the REPL, it seems like you  could compare tuples instead: `ZOAU_API_VERSION > (1, 2, 2)`. This way if `ZOAU_API_VERSION` was something like 1.2.xx, it would be evaluated as bigger.

It's probably not really needed because ZOAU plans to release 1.3.0 relatively soon, but I guess it's good to keep it in mind just in case.

Reply:
good grief... that's a newbie mistake, and I made it!  Good catch!

Reply:
I re-ran it and realized this case would work correctly.  The feature is added in 1.2.3, so unless they backport the change, it SHOULD failed on 1.2.anything.  1.2.10 is not > 1.2.2